frontend/packages/topology/src/components/graph-view/components/nodes/WorkloadNode.tsx

```diff
@@ -13,18 +15,134 @@ import {
 } from '@patternfly/react-topology';
 import { useTranslation } from 'react-i18next';
 import { getImageForIconClass } from '@console/internal/components/catalog/catalog-item-icon';
-import { calculateRadius, PodRCData, usePodsWatcher } from '@console/shared';
+import { AlertSeverity } from '@console/internal/components/monitoring/types';
+import {
+  AllPodStatus,
+  calculateRadius,
+  getFiringAlerts,
+  getPodStatus,
+  getSeverityAlertType,
+  PodRCData,
+  useBuildConfigsWatcher,
+  usePodsWatcher,
+} from '@console/shared';
 import { getFilterById, SHOW_POD_COUNT_FILTER_ID, useDisplayFilters } from '../../../../filters';
-import { getTopologyResourceObject } from '../../../../utils/topology-utils';
+import { getResource, getTopologyResourceObject } from '../../../../utils/topology-utils';
 import BaseNode from './BaseNode';
 import { getNodeDecorators } from './decorators/getNodeDecorators';
 import PodSet, { podSetInnerRadius } from './PodSet';
 
 import './WorkloadNode.scss';
 
+const POD_STATUS_NORMAL = 1;
+const POD_STATUS_WARNING = 2;
+const POD_STATUS_DANGER = 3;
+
+const buildStatusSeverities = {
+  danger: [
+    'ContainerCannotRun',
+    'CrashLoopBackOff',
+    'Critical',
+    'ErrImagePull',
+    'Error',
+    'Failed',
+    'Failure',
+    'ImagePullBackOff',
+    'InstallCheckFailed',
+    'Lost',
+    'Rejected',
+    'UpgradeFailed',
+  ],
+  warning: [
+    'Cancelled',
+    'Deleting',
+    'Expired',
+    'Not Ready',
+    'Terminating',
+    'Warning',
+    'RequiresApproval',
+  ],
+  normal: [
+    'New',
+    'Pending',
+    'Planning',
+    'ContainerCreating',
+    'UpgradePending',
+    'In Progress',
+    'Installing',
+    'InstallReady',
+    'Replacing',
+    'Running',
+    'Updating',
+    'Upgrading',
+    'Cancelled',
+    'Deleting',
+    'Expired',
+    'Not Ready',
+    'Terminating',
+    'Accepted',
+    'Active',
+    'Bound',
+    'Complete',
+    'Completed',
+    'Created',
+    'Enabled',
+    'Succeeded',
+    'Ready',
+    'Up to date',
+    'Provisioned as node',
+    'Preferred',
+    'Connected',
+    'Info',
+    'Unknown',
+    'PipelineNotStarted',
+  ],
+};
```

Comment:
@karthikjeeyar Could you verify these build status categories? They are based on https://github.com/openshift/console/blob/master/frontend/packages/console-dynamic-plugin-sdk/src/app/components/status/Status.tsx#L32